thirdparty/nginx/nginx-1.7.1-zimbra/src/core/ngx_zm_lookup.c

```diff
@@ -1981,8 +1991,9 @@ ngx_zm_lookup_cache_route(ngx_zm_lookup_ctx_t *ctx, ngx_str_t user, ngx_str_t ro
     work = ctx->work;
 
     if (work->auth_method == ZM_AUTHMETH_ZIMBRAID) {
-            key = ngx_zm_lookup_get_id_route_key(ctx->pool, log,
-                    ZM_PROTO[work->protocol], user, work->isAdmin);
+            key = ngx_zm_lookup_get_id_route_key(
+                    ctx->pool, log,
+                    ZM_PROTO[work->protocol], user, work->isAdmin, work->isZx);
```

Comment:
Could it be explained about work->isAdmin here, as we are replacing it with work->isZx. But from above code, we are still checking work->isAdmin in some places.

Reply:
It is always better to pass one of these, rather than passing both.